pkg/ast/spl/spl.peg

```diff
@@ -951,6 +956,301 @@ TcOptionCMD <- option:("usenull" / "useother" / "nullstr" / "otherstr") {
     return optionStr, nil
 }
 
+SpanLenOption <- number:(FloatAsString / IntegerAsString) timeScale:(TimeScale)? {
+    if timeScale != nil {
+        num, err := strconv.ParseInt(number.(string), 10, 64)
+        if err != nil {
+            return nil, fmt.Errorf("SpanLength: Invalid num (%v): %v", number.(string), err)
+        }
+
+        switch timeScale.(utils.TimeUnit) {
+            case utils.TMDecisecond:
+                if 10%num != 0 || num >= 10 {
+                    return nil, fmt.Errorf("SpanLength: Invalid span for decisecond %v, span value must be less than 1 second and 1 second must be evenly divisble by span", num)
+                }
+            case utils.TMCentisecond:
+                if 100%num != 0 || num >= 100 {
+                    return nil, fmt.Errorf("SpanLength: Invalid span for centisecond %v, span value must be less than 1 second and 1 second must be evenly divisble by span", num)
+                }
+            case utils.TMMillisecond:
+                if 1000%num != 0 || num >= 1000 {
+                    return nil, fmt.Errorf("SpanLength: Invalid span for millisecond %v, span value must be less than 1 second and 1 second must be evenly divisble by span", num)
+                }
+        }
+
+        spanLength := &structs.BinSpanLength {
+            Num: float64(num),
+            TimeScale: timeScale.(utils.TimeUnit),
+        }
+
+        return spanLength, nil
+    } 
+
+    num, err := strconv.ParseFloat(number.(string), 64)
+    if err != nil {
+        return nil, fmt.Errorf("SpanLength: Invalid num (%v): %v", number.(string), err)
+    }
+
+    spanLength := &structs.BinSpanLength {
+        Num: num,
+        TimeScale: utils.TimeUnit(0),
+    }
+
+    return spanLength, nil
+}
+
+BinOptionMinSpan <- "minspan" EQUAL spanLength:SpanLenOption {
+
+    binOptionArgs := &BinOptionArgs {
+        argOption: "minspan",
+        binOptions: &structs.BinCmdOptions {
+            MinSpan: spanLength.(*structs.BinSpanLength),
+        },
+    }
+    return binOptionArgs, nil
+}
+
+BinOptionMaxBins <- "bins" EQUAL intAsStr:IntegerAsString {
+    numBins, err := strconv.ParseUint(intAsStr.(string), 10, 64)
+    if err != nil {
+        return nil, fmt.Errorf("Invalid limit (%v): %v", intAsStr.(string), err)
+    }
+    binOptionArgs := &BinOptionArgs {
+        argOption: "bins",
+        binOptions: &structs.BinCmdOptions {
+            MaxBins: numBins,
+        },
+    }
+    return binOptionArgs, nil
+}
+
+PositiveIntegerAsString <- [0-9]+ {
+    return string(c.text), nil
+}
```

Comment:
nit: move this near the `IntegerAsString` rule

Reply:
addressed in the latest commit.